plugins/nexus-repository-manager/src/api/nexus-repository-manager-api-client/nexus-repository-manager-api-client.ts

```diff
@@ -74,11 +87,15 @@ export class NexusRepositoryManagerApiClient
     return await SearchService.search(query);
   }
 
-  private async fetcher(url: string) {
+  private async fetcher(
+    url: string,
+    additionalHeaders: Record<string, string> = {},
```

Comment:
I am not sure if this is the right level of abstraction (should this be `accept: string = "application/json"`), the right type (maybe [`Headers`](https://developer.mozilla.org/en-US/docs/Web/API/Headers)?), or the idiomatic way to do optional arguments (`additionalHeaders?: Record<string, string>` with `...(additionalHeaders ?? {})`).

Feedback here would be appreciated.

Reply:
Here is how I would update the `fetcher` method.

```ts
private async fetcher(url: string, additionalHeaders: HeadersInit = {}) {
  const { token: idToken } = await this.identityApi.getCredentials();

  const headers = new Headers(additionalHeaders);
  headers.set('Content-Type', 'application/json');
  if (idToken) {
    headers.set('Authorization', `Bearer ${idToken}`);
  }

  const response = await fetch(url, {
    headers,
  });
  if (!response.ok) {
    throw new Error(
      `failed to fetch data, status ${response.status}: ${response.statusText}`,
    );
  }
  return await response.json();
}
```